### PR TITLE
gh-101819: Adapt _io.IOBase.seek and _io.IOBase.truncate to Argument Clinic

### PR DESCRIPTION
**Pull request by erlend-aasland**
Pass defining class, so we can easily fetch module state in the future.

* Issue: gh-101819

### CODE REVIEW

**Review by kumaraditya303 — APPROVED** [review]

**Comment by corona10** [comment]
@erlend-aasland 
Out of curiosity, what are the criteria for converting to AC tools? 
I love to use AC tool if possible and we have a bunch of methods to convert but I am not sure when to do when to not.

**Comment by erlend-aasland** [comment]
For this particular change, I need the defining class in the near future. This PR is a prerequisite for the complete isolation of \_io.

Also, the Faster Python team is pushing AC adaption (see open issue in their repo), as it will make it easier to experiment with class method optimisations.

**Comment by erlend-aasland** [comment]
See also https://github.com/faster-cpython/ideas/issues/565

**Comment by corona10** [comment]
> For this particular change, I need the defining class in the near future

+1

> Also, the Faster Python team is pushing AC adaption (see open issue in their repo), 

Yeah, I sometimes observe manual optimization patches that decide not to use AC.
Such decisions make me confusion..

File: Modules/_io/iobase.c [hunk 1]
```diff
@@ -79,21 +79,27 @@ iobase_unsupported(_PyIO_State *state, const char *message)
...
-    "Change the stream position to the given byte offset. The offset is\n"
-    "interpreted relative to the position indicated by whence.  Values\n"
-    "for whence are:\n"
-    "\n"
-    "* 0 -- start of stream (the default); offset should be zero or positive\n"
-    "* 1 -- current stream position; offset may be negative\n"
-    "* 2 -- end of stream; offset is usually negative\n"
-    "\n"
-    "Return the new absolute position.");
+/*[clinic input]
```
Comment (vstinner) [inline Modules/_io/iobase.c:82]: Usually there is an empty line before the list of arguments. It seems like it's not required. I'm not aware of coding style for argument clinic.

**Review by vstinner — COMMENTED** [review]
Review after merge: LGTM. I just left a minor coding style remark.

Thanks for spliting your large draft PR into smaller PRs. It makes reviews easier, but also makes the Git history easier to follow.

File: Modules/_io/iobase.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_io/iobase.c:82]: Both styles are present in the code base, and there is no consistant style in Modules/_io. For sqlite3, I think all clinic code have the extra newline.

**Comment by erlend-aasland** [comment]
Thanks for the post-merge reviews! There are still a number of PRs left, so if there are remarks, we can easily address those post merge. Most of the low-hanging fruit is now picked; soon we'll get to porting the base types! It gets interesting.

File: Modules/_io/iobase.c (hunk 1 above)
Comment (vstinner) [reply Modules/_io/iobase.c:82]: My reference is the os module. I'm fine with having both coding styles until Argument Clinic becomes more strict or a coding style is defined (ex: in PEP 7).

**Pull request merged**